sdk/resource/resource.go

```diff
@@ -177,7 +204,9 @@ func Merge(a, b *Resource) (*Resource, error) {
 	case a.schemaURL == b.schemaURL:
 		schemaURL = a.schemaURL
 	default:
-		return Empty(), errMergeConflictSchemaURL
+		// Return the merged resource with an appropriate error. It is up to
+		// the user to decide if the returned resource can be used or not.
+		err = ErrSchemaURLConflict
```

Comment:
It might be a good idea to incorporate the idea behind #4857 and include the schemas that are in conflict.

Reply:
I was thinking about the same, but then we would have to change `ErrSchemaURLConflict` to be a type (instead of value). Checking for error using [`errors.As`](https://pkg.go.dev/errors#example-As) requires more boilerplate. I though that simplicity may be better.

Reply:
I think that PR does a good job at handling the scope it is tackling. I was planning to let it handle that change there and not increase the scope of this PR.

Reply:
Hmm, I guess they haven't signed the CLA yet there ... I'll add that work here.

Reply:
Updated.